### PR TITLE
Add Debezium chaos testing

### PR DESCRIPTION
**Pull request by JLDLaughlin**
This PR allows us to chaos test the `mysql -> debezium -> materialize` pipeline with the following new tests:
- pausing and unpausing the `debezium/connect` container
- stopping and starting the container
- killing the container
- tc netem chaos tests: delay, rate, loss, duplicate, corrupt

To do this, I've added the requisite `chbench` Docker services (`debezium/connect`, a tppch-specific `connector`, `mysql`, and our `chbench` service) and a new `mysql_debezium_kafka` chaos test function.

I tested each new test locally, once I merge them I will spin them up on EC2.

This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/3400)

### CODE REVIEW

File: test/chaos/connect/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+FROM debezium/connect:1.0
+
+# Reset the user to root in order to update the container's networking
+# configurations via tc netem.
+USER root
```
Comment (JLDLaughlin) [inline test/chaos/connect/Dockerfile:14]: https://www.youtube.com/watch?v=O2yPnnDfqpw

File: test/chaos/src/main.rs [hunk 2]
```diff
@@ -30,15 +31,86 @@ async fn run() -> Result<(), anyhow::Error> {
...
+    mz_client.execute(&*qo1_query, &[]).await?;
+
+    // Query both materialized views until run_seconds has expired, raise any errors.
+    while std::time::Instant::now() < end {
+        let count = "SELECT * FROM orderline_count;";
+        mz_client::try_query_one(&mz_client, &*count, Duration::from_secs(1)).await?;
+
+        let q01 = "SELECT * FROM q01;";
+        mz_client::try_query(&mz_client, &q01, Duration::from_secs(1)).await?;
+    }
```
Comment (quodlibetor) [inline test/chaos/src/main.rs:97]: this is good, but I think going forward this should probably share some of the validation code with checker. I think you've already said that you would like to share the code in `test_utils` or something, just calling it out.

Comment (JLDLaughlin) [reply test/chaos/src/main.rs:97]: yeah agreed!

File: test/chaos/mzcompose.yml [hunk 3]
```diff
@@ -241,3 +465,60 @@ mzconduct:
...
+
+    # chbench helper workflows
+    chbench-chaos-test:
+      steps:
+        - step: workflow
+          workflow: chbench-bring-up-source-data
+        # start the dashboard before the load data get started so that we have data from
+        # the beginning of time
+        - step: start-services
+          services: [dashboard]
```
[the remark this answers is not in the document]
Comment (JLDLaughlin) [reply test/chaos/mzcompose.yml:477]: oops -- yes!

File: test/chaos/connector/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+RUN sed -i s,https://s3-us-west-2.amazonaws.com/staging-confluent-packages-5.3.0/deb/5.3,https://packages.confluent.io/deb/5.3, /etc/apt/sources.list
+
+RUN apt-get update && apt-get -qy install curl
+
+RUN curl -fsSL https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh > /usr/local/bin/wait-for-it \
+    && chmod +x /usr/local/bin/wait-for-it
+
+COPY docker-entrypoint.sh /usr/local/bin
+
+ENTRYPOINT ["docker-entrypoint.sh"]
```
[the remark this answers is not in the document]
Comment (JLDLaughlin) [reply test/chaos/connector/Dockerfile:22]: no -- good catch! I copied this over from `chbench`, but I will switch it around.

File: misc/python/materialize/cli/mzconduct.py [hunk 5]
```diff
@@ -949,108 +954,85 @@ def run(self, comp: Composition, workflow: Workflow) -> None:
...
-    def __init__(self, container: str, percent: int, duration: int = -1) -> None:
-        super().__init__(duration=duration)
+    def __init__(self, container: str, percent: int) -> None:
         self._container = container
         self._percent = percent
 
-    def get_cmd(self, duration: int) -> str:
-        return f"pumba netem --duration {duration}m loss --percent {self._percent} {self._container}"
+    def get_cmd(self) -> str:
+        return f"docker exec -it {self._container} tc qdisc add dev eth0 root netem loss {self._percent}"
```
[the remark this answers is not in the document]
Comment (JLDLaughlin) [reply misc/python/materialize/cli/mzconduct.py:1009]: (in other comment!)

**Comment by JLDLaughlin** [comment]
So to:
> But also you might be able to do this by adding the NET_ADMIN or something capability, instead? https://stackoverflow.com/q/58377469/25616

We do add the `NET_ADMIN` capability to the `connect` container. But, the underlying image of the `connect` container ([`debezium/kafka`](https://hub.docker.com/r/debezium/kafka/dockerfile)) sets the user to `KAFKA`, which doesn't have permissions to update network configurations. So even with `NET_ADMIN`, running `tc netem...` was failing.

In the same vein:
> why is this switching from pumba to tc? This seems reasonable, and like it's basically doing the same thing, but I'm not knowledgable about this, curious if it's fixing something or just different.

Even with the `connect` container running as root with the `NET_ADMIN` capability, pumba was still failing. I didn't spend too much time digging into it, but I did realize I could get rid of the bulky blocking/non-blocking stuff if I just used `tc` directly anyway.

**Comment by quodlibetor** [comment]
Cool, that's unfortunate but works for me! +2

**Pull request merged**